File: cs50-rust/ps1-01-deep-thought/src/main.py
def main():
   great_answer_life = ["42", "forty two", "forty-two", "my love caron"]
   user_input = input("What is the Great Question of Life, the Universe and Everything?\n")
   if user_input.strip().lower() in great_answer_life:
       print("Yes")
   else:
       print("No")

main()

File: cs50-rust/ps1-01-deep-thought/src/main.rs
use std::io::stdin;

fn main() {
    println!();
    let great_answer_life: [&str; 4] = ["42", "forty-two", "forty two", "my love caron"];

    println!("What is the Answer to the Great Question of Life, the Universe, and Everything?");
    let mut user_input = String::new();
    stdin()
        .read_line(&mut user_input)
        .expect("Error reading line");

    let input_lowered = user_input.trim().to_lowercase();

    if great_answer_life.contains(&input_lowered.as_str()) {
        println!("Yes")
    } else {
        println!("No")
    }
}
